pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -475,45 +475,51 @@ func buildInboundHTTPFilter(mesh *meshconfig.MeshConfig, attrs attributes, node
 	return out
 }
 
-func addFilterConfigToRoute(in *plugin.InputParams, httpRoute *route.Route, attrs attributes) {
+func addFilterConfigToRoute(in *plugin.InputParams, httpRoute *route.Route, attrs attributes,
+	quotaSpec []*mccpb.QuotaSpec) {
 	httpRoute.TypedPerFilterConfig = addTypedServiceConfig(httpRoute.TypedPerFilterConfig, &mccpb.ServiceConfig{
 		DisableCheckCalls:  disablePolicyChecks(outbound, in.Push.Mesh, in.Node),
 		DisableReportCalls: in.Push.Mesh.GetDisableMixerHttpReports(),
 		MixerAttributes:    &mpb.Attributes{Attributes: attrs},
 		ForwardAttributes:  &mpb.Attributes{Attributes: attrs},
+		QuotaSpec:          quotaSpec,
 	})
 }
 
 func modifyOutboundRouteConfig(push *model.PushContext, in *plugin.InputParams, virtualHostname string, httpRoute *route.Route) *route.Route {
+	isPolicyCheckDisabled := disablePolicyChecks(outbound, in.Push.Mesh, in.Node)
 	// default config, to be overridden by per-weighted cluster
 	httpRoute.TypedPerFilterConfig = addTypedServiceConfig(httpRoute.TypedPerFilterConfig, &mccpb.ServiceConfig{
-		DisableCheckCalls:  disablePolicyChecks(outbound, in.Push.Mesh, in.Node),
+		DisableCheckCalls:  isPolicyCheckDisabled,
 		DisableReportCalls: in.Push.Mesh.GetDisableMixerHttpReports(),
 	})
 	switch action := httpRoute.Action.(type) {
 	case *route.Route_Route:
 		switch upstreams := action.Route.ClusterSpecifier.(type) {
 		case *route.RouteAction_Cluster:
-			_, _, hostname, _ := model.ParseSubsetKey(upstreams.Cluster)
-			var attrs attributes
-			if hostname == "" && upstreams.Cluster == util.PassthroughCluster {
-				attrs = addVirtualDestinationServiceAttributes(make(attributes), util.PassthroughCluster)
-			} else {
-				svc := in.Node.SidecarScope.ServiceForHostname(hostname, push.ServiceByHostnameAndNamespace)
-				attrs = addDestinationServiceAttributes(make(attributes), svc)
+			{
```

Comment:
donot change this, it makes hard to review what you changed

Reply:
Sure, removed {}